kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -219,6 +267,105 @@
+        };
+        return ContentNodeProgressResource.fetchCollection({ getParams }).then(progresses => {
+          this.viewResourcesContents = nodes
+            .map(node => {
+              const matchingProgress = progresses.find(p => p.content_id === node.content_id) || {
+                progress: 0,
+              };
+              node.progress = matchingProgress.progress;
+              return node;
+            })
+            .filter(node => node.content_id !== this.content.content_id);
+        });
+      },
+      /**
+       * Prepares a list of content nodes which are also in the same lesson as this.lessonId
+       * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
+       */
+      fetchLessonSiblings() {
+        LessonResource.fetchModel({ id: this.lessonId }).then(lesson => {
+          ContentNodeResource.fetchCollection({
+            getParams: {
+              ids: lesson.resources.map(resource => resource.contentnode_id),
+            },
+          }).then(contentNodes => {
+            return this.progressPromise(contentNodes);
+          });
+        });
+      },
+      /**
+       * Prepares a list of content nodes which are children of this.content.parent without
+       * this.content. The prepared nodes have their progress values for the current user
+       * (if there is one) mapped onto each node.
+       *
+       * Largely borrowed from modules/topicsTree/handlers.js
+       *
+       * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
+       * @modifies this.nextFolder - Sets the value with this.content's parents next sibling folder
+       * if found
+       */
+      fetchSiblings() {
+        // Guard against this in lesson context
+        if (this.lessonId) {
+          return;
+        }
+
+        ContentNodeResource.fetchTree({
+          id: this.content.parent,
+          params: {
+            include_coach_content:
+              this.$store.getters.isAdmin ||
+              this.$store.getters.isCoach ||
+              this.$store.getters.isSuperuser,
+          },
+        }).then(parent => {
+          // Filter out this.content
+          const nodes = parent.children.results.filter(node => node.id !== this.content.id);
+
+          /** A promise to get the nextFolder content - done always */
+          const nextFolderPromise = ContentNodeResource.fetchNextContent(parent.id, {
+            topicOnly: true,
+          }).then(nextContent => {
+            // This may return the immediate parent if nothing else is found so let's be sure
+            // not to assign that
+            if (nextContent && this.content.parent !== nextContent.id) {
+              this.nextContent = nextContent;
+            }
+          });
+
+          const promises = [nextFolderPromise];
+          // Push progressPromise, which handles user logged in logic
+          promises.push(this.progressPromise(nodes));
+          Promise.all(promises);
```

Comment:
Doesn't look like you are actually returning `Promise.all` here - maybe don't need to track the promises at all? Can just fire them off and let their changes in state do the talking.